lightning/src/chain/chaininterface.rs

```diff
@@ -252,12 +253,21 @@ impl<'a, CL: Deref<Target = ChainListener + 'a> + 'a> BlockNotifier<'a, CL> {
 	}
 
 	/// Register the given listener to receive events.
-	// TODO: unregister
 	pub fn register_listener(&self, listener: CL) {
 		let mut vec = self.listeners.lock().unwrap();
 		vec.push(listener);
 	}
 
+    /// Unregister the given listener to no longer
+    /// receive events.
+    ///
+    /// If the same listener is registered multiple times, unregistering
+    /// will remove ALL occurrences of that listener.
+    pub fn unregister_listener(&self, listener: CL) {
+        let mut vec = self.listeners.lock().unwrap();
+        vec.retain(|item| !ptr_eq(&(**item) as *const _, &(*listener) as *const _));
```

Comment:
Can you add a brief comment just saying "item is a ref to an abstract thing that deferences to a ChainListener, so dereference it twice to get the ChainListener itself". Note that I don't believe you need the "as *const _" bits.